src/Explorer/Menus/CommandBar/MemoryTrackerComponent.tsx

```diff
@@ -1,50 +1,71 @@
-import * as React from "react";
-import { Observable, Subscription } from "knockout";
-import { MemoryUsageInfo } from "../../../Contracts/DataModels";
+import React, { FunctionComponent } from "react";
+import useSWR from "swr";
 import { ProgressIndicator } from "office-ui-fabric-react/lib/ProgressIndicator";
 import { Spinner, SpinnerSize } from "office-ui-fabric-react/lib/Spinner";
 import { Stack } from "office-ui-fabric-react/lib/Stack";
+import { listConnectionInfo } from "../../../Utils/arm/generatedClients/2020-04-01-notebooks/notebookWorkspaces";
+import { NotebookWorkspaceConnectionInfoResult } from "../../../Utils/arm/generatedClients/2020-04-01-notebooks/types";
+import { userContext } from "../../../UserContext";
 
-interface MemoryTrackerProps {
-  memoryUsageInfo: Observable<MemoryUsageInfo>;
+export interface MemoryUsageInfo {
+  total: number;
+  free: number;
 }
 
-export class MemoryTrackerComponent extends React.Component<MemoryTrackerProps> {
-  private memoryUsageInfoSubscription: Subscription;
+const kbInGB = 1048576;
 
-  public componentDidMount(): void {
-    this.memoryUsageInfoSubscription = this.props.memoryUsageInfo.subscribe(() => {
-      this.forceUpdate();
-    });
-  }
-
-  public componentWillUnmount(): void {
-    this.memoryUsageInfoSubscription && this.memoryUsageInfoSubscription.dispose();
-  }
-
-  public render(): JSX.Element {
-    const memoryUsageInfo: MemoryUsageInfo = this.props.memoryUsageInfo();
-    if (!memoryUsageInfo) {
-      return (
-        <Stack className="memoryTrackerContainer" horizontal>
-          <span>Memory</span>
-          <Spinner size={SpinnerSize.medium} />
-        </Stack>
-      );
+const fetchMemoryInfo = async (_key: unknown, connectionInfo: NotebookWorkspaceConnectionInfoResult) => {
+  const response = await fetch(`${connectionInfo.notebookServerEndpoint}/api/metrics/memory`, {
+    method: "GET",
+    headers: {
+      Authorization: `Token ${connectionInfo.authToken}`,
+      "content-type": "application/json"
     }
+  });
+  if (!response.ok) {
+    throw new Error(await response.text());
```

Comment:
We should also try to restart the notebook server if the call fails. See `onConnectionLost` in NotebookContainerClient.